scalopus_examples/src/showcase_count_events.cpp

```diff
@@ -27,62 +27,48 @@
   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
-#include "scalopus_tracing/tracing.h"
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <thread>
 
-int main(int /* argc */, char** /* argv */)
-{
-  // Execute all public trace point macros.
-  // Mostly to ensure they don't collide with each other if used multiple times
-  // or create shadowed variables or something.
-  TRACE_PRETTY_FUNCTION();
-  TRACE_PRETTY_FUNCTION();
-
-  TRACE_SCOPE_RAII("main");
-  TRACE_SCOPE_RAII("main");
-
-  TRACE_SCOPE_START("zz");
-  TRACE_SCOPE_START("zz");
-  TRACE_SCOPE_END("zz");
-  TRACE_SCOPE_END("zz");
+#include <scalopus_tracing/tracing.h>
+#include <scalopus_transport/transport_unix.h>
 
-  TRACING_CONFIG_THREAD_STATE_RAII(true);
-  TRACING_CONFIG_THREAD_STATE_RAII(true);
+// This example shows how counters can be used.
 
-  TRACING_CONFIG_PROCESS_STATE_RAII(true);
-  TRACING_CONFIG_PROCESS_STATE_RAII(true);
-
-  TRACE_MARK_EVENT_GLOBAL("foo");
-  TRACE_MARK_EVENT_GLOBAL("foo");
-  TRACE_MARK_EVENT_PROCESS("bar");
-  TRACE_MARK_EVENT_PROCESS("bar");
-  TRACE_MARK_EVENT_THREAD("buz");
-  TRACE_MARK_EVENT_THREAD("buz");
-
-  // Check if the macro's don't expand to shadowed variables.
+int main(int /* argc */, char** argv)
+{
+  auto factory = std::make_shared<scalopus::TransportUnixFactory>();
+  const auto server = factory->serve();
+  server->addEndpoint(std::make_unique<scalopus::EndpointTraceMapping>());
+  server->addEndpoint(std::make_unique<scalopus::EndpointIntrospect>());
+  // Native sender is not needed for LTTng or NOP tracepoints, but it must be present if the native tracepoints are
+  // ever to be used.
+  server->addEndpoint(std::make_shared<scalopus::EndpointNativeTraceSender>());
+  auto endpoint_process_info = std::make_shared<scalopus::EndpointProcessInfo>();
+
+  // Set the process name here:
+  endpoint_process_info->setProcessName(argv[0]);
+  server->addEndpoint(endpoint_process_info);
+
+  // Set the thread name.
+  TRACE_THREAD_NAME("main");
+
+  double t{ 0 };
+  int steps{ 0 };
+  while (true)
   {
-    TRACE_PRETTY_FUNCTION();
-    TRACE_PRETTY_FUNCTION();
-
-    TRACE_SCOPE_RAII("main");
-    TRACE_SCOPE_RAII("main");
-
-    TRACE_SCOPE_START("zz");
-    TRACE_SCOPE_START("zz");
-    TRACE_SCOPE_END("zz");
-    TRACE_SCOPE_END("zz");
-
-    TRACING_CONFIG_THREAD_STATE_RAII(true);
-    TRACING_CONFIG_THREAD_STATE_RAII(true);
-
-    TRACING_CONFIG_PROCESS_STATE_RAII(true);
-    TRACING_CONFIG_PROCESS_STATE_RAII(true);
-
-    TRACE_MARK_EVENT_GLOBAL("foo");
-    TRACE_MARK_EVENT_GLOBAL("foo");
-    TRACE_MARK_EVENT_PROCESS("bar");
-    TRACE_MARK_EVENT_PROCESS("bar");
-    TRACE_MARK_EVENT_THREAD("buz");
-    TRACE_MARK_EVENT_THREAD("buz");
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    steps++;
+    t += 0.1;
+    std::int64_t a = static_cast<std::int64_t>(100.0 * std::sin(t * 10.0) + 100.0);
```

Comment:
pedantic: rename `a` and `b` to `slow` and `fast` and use the same order before in the `TRACE_COUNT_SERIES`

Reply:
Yes, that's better.